### PR TITLE
mydumper: fix parquet data parser

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
According to [Parquet Logical Type Definitions](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md) and [parquet-go](https://github.com/xitongsys/parquet-go), the value for some logic types (decimal, date, datetime, timestamp) store in parquet files may not be compatible with mysql/tidb value. We must do a manual convert in our reader.

### What is changed and how it works?
Do convert these types' value to string according parquet and tidb definition.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

File: lightning/mydump/parquet_parser_test.go [hunk 1]
```diff
@@ -6,11 +6,11 @@ import (
 	"path/filepath"
 	"strconv"
 
-	"github.com/pingcap/br/pkg/storage"
-
-	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/parser/mysql"
```
Comment (overvenus) [inline lightning/mydump/parquet_parser_test.go:10]:
```suggestion
```

File: lightning/mydump/parquet_parser.go [hunk 2]
```diff
@@ -166,33 +172,88 @@ func (pp *ParquetParser) ReadRow() error {
...
 	case reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
 		d.SetUint64(v.Uint())
-	case reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+	case reflect.Int8, reflect.Int16:
 		d.SetInt64(v.Int())
+	case reflect.Int32, reflect.Int64:
+		setDatumByInt(d, v.Int(), meta)
 	case reflect.String:
-		d.SetString(v.String(), "")
+		d.SetString(v.String(), mysql.DefaultCollationName)
```
Comment (overvenus) [inline lightning/mydump/parquet_parser.go:192]: Is `mysql.DefaultCollationName` fit for all scenarios?

File: lightning/mydump/parquet_parser.go [hunk 3]
```diff
@@ -185,14 +195,64 @@ func setDatumValue(d *types.Datum, v reflect.Value) {
...
+	}
+	switch *meta.ConvertedType {
+	// decimal
+	case parquet.ConvertedType_DECIMAL:
+		scale := int64(1)
+		for i := 0; i < int(*meta.Scale); i++ {
+			scale *= 10
+		}
+		fmtStr := fmt.Sprintf("%%d.%%0%dd", *meta.Scale)
+		val := fmt.Sprintf(fmtStr, v/scale, abs(v%scale))
```
Comment (kennytm) [inline lightning/mydump/parquet_parser.go:227]:
```suggestion
		val := fmt.Sprintf("%d.%0*d", v/scale, *meta.Scale, abs(v%scale))
```

but maybe it's more efficient to just print `fmt.Sprintf("%0*d", *meta.Scale + N, v)` (for a suitable value of N) and then insert a `.` to the resulting string.

File: lightning/mydump/parquet_parser_test.go [hunk 4]
```diff
@@ -81,3 +79,69 @@ func (s testParquetParserSuite) TestParquetParser(c *C) {
...
+
+	v := &Test{
+		Date:            18564,              //2020-10-29
+		TimeMillis:      62775123,           // 17:26:15.123
+		TimeMicros:      62775123000,        // 17:26:15.123
+		TimestampMillis: 1603963672356,      // 2020-10-29T17:27:52.356
+		TimestampMicros: 1603963672356956,   //2020-10-29T17:27:52.356956
+		Decimal1:        -12345678,          // -123456.78
+		Decimal2:        456,                // 0.0456
+		Decimal3:        123456789012345678, //1234567890123456.78
```
Comment (kennytm) [inline lightning/mydump/parquet_parser_test.go:116]: add a test case for these decimals:

* `0.000`
* `1.000`
* `-1.000`
* `0.999`
* `-0.999`
* `-0.0001`

**Comment by glorv** [comment]
@kennytm  @overvenus PTAL, again

**Review by kennytm — COMMENTED** [review]
LGTM

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**